### PR TITLE
Basketball game information

### PR DESCRIPTION
**Pull request by EdvinasA**
Did all the tasks from first to last,but haven't implemented game stats, but will try to do it before the weekend, also i will shorten the code and do more reusing, but for the first pull request showing my first version of code.

### CODE REVIEW

File: src/main/java/BasketballGame/GameRegister.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+import static java.time.DayOfWeek.SUNDAY;
+import static java.time.temporal.TemporalAdjusters.nextOrSame;
+import static java.time.temporal.TemporalAdjusters.previousOrSame;
+
+public class GameRegister {
+    //I know we didn't lear anything about ArrayList, but i have did that on my own and for
+    // it's a lot easier to do it with ArrayList.
+    private final ArrayList<Game> game;
+    private LocalDateTime localDateTime = LocalDateTime.now();
+    private final DateTimeFormatter myFormatObj = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
```
Comment (EdvinasA) [inline src/main/java/BasketballGame/GameRegister.java:17]: Added localDateTime and DateTimeFormatter

**Comment by Almantask** [comment]
image
Try to name commits a bit better 😄

File: src/main/java/BasketballGame/Coach.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    }
+
+    public String getCoachTitle() {
+        return "Coach";
+    }
+
+    public Human getCoach() {
+        return coach;
+    }
+}
```
Comment (Almantask) [inline src/main/java/BasketballGame/Coach.java]: Whole class can be removed. Coach has nothing of its' own.

File: src/main/java/BasketballGame/DateTime.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
+package BasketballGame;
+
+public class DateTime {
```
Comment (Almantask) [inline src/main/java/BasketballGame/DateTime.java:3]: I actually made a mistake, could have used LocalDateTime instead.

File: src/main/java/BasketballGame/Game.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+    public Game(Team teamOne, int teamOneScore, Team teamTwo, int teamTwoScore, LocalDateTime gameTime) {
+        this.teamOne = teamOne;
+        this.teamTwo = teamTwo;
+        this.teamOneScore = teamOneScore;
+        this.teamTwoScore = teamTwoScore;
+        this.gameTime = gameTime;
+    }
+
+    public static String getWinner(Game game) {
```
Comment (Almantask) [inline src/main/java/BasketballGame/Game.java:24]: What if I wanted to print a different message, but consistently in both cases? Avoid doing more than is asked- all that is asked is to return the winning team.
It should return the winning team or "Draw", nothing more.

File: src/main/java/BasketballGame/GameRegister.java [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+        System.out.println("|" + fixedLengthString(game.getTeamOne().getCoach().getCoachTitle(),22) + "|" + fixedLengthString(game.getTeamOne().getCoach().getCoach().getName(),22) + "|    |       |    |"
+                + fixedLengthString(game.getTeamTwo().getCoach().getCoachTitle(),24) + "|" + fixedLengthString(game.getTeamTwo().getCoach().getCoach().getName(),21) + "|");
+        for (int i = 0; i < teamOne.length; i++) {
+            System.out.println("|"+fixedLengthString(String.valueOf(teamOne[i]),22)+"|" + fixedLengthString(teamOneNames[i],22)+"|    |       |    |"
+                    + fixedLengthString(String.valueOf(teamTwo[i]),24)+"|"+fixedLengthString(teamTwoNames[i], 21)+"|");
+        }
+        System.out.println("|----------------------|----------------------|----|-------|----|------------------------|---------------------|");
+        System.out.println("Match started at: " + game.getGameTime().format(myFormatObj));
+
+    }
```
Comment (Almantask) [inline src/main/java/BasketballGame/GameRegister.java]: Split this code into steps:
GenerateTeam(String name);
team.ToString()
PrintRow(team.ToString())
PrintTopOrBot()

**Review by Almantask — CHANGES_REQUESTED** [review]

File: src/main/java/BasketballGame/Game.java (hunk 4 above)
Comment (EdvinasA) [reply src/main/java/BasketballGame/Game.java:24]: Now it returns only The winner or Draw

File: src/main/java/BasketballGame/GameRegister.java [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+                printDate(i);
+            }
+        }
+    }
+
+    public void printGameInformation (Game game) {
+        topOrBottom(game,"Top");
+        printRow(game);
+        topOrBottom(game,"Bottom");
+    }
```
Comment (EdvinasA) [inline src/main/java/BasketballGame/GameRegister.java:92]: Don't know if i understood correctly, but I guess this kinda looks what you asked for printing game information (Hard part of the Challenge).

**Review by EdvinasA — COMMENTED** [review]
Deleted Coach class,also should i leave DateTime or just create like you said with LocalDateTime ?

**Pull request closed without merging**